sklearn/cluster/k_means_.py

```diff
@@ -1282,15 +1289,19 @@ def fit(self, X, y=None):
             Coordinates of the data points to cluster
         """
         random_state = check_random_state(self.random_state)
-        X = check_array(X, accept_sparse="csr", order='C', dtype=np.float64)
+        if sp.issparse(X):
```

Comment:
Remind me why we need this conditional, i.e. why `check_array(X, accept_sparse="csr", order='C', dtype=[np.float64, np.float32])` is insufficient?